[Alpha][13.0] Add stock_cubiscan

guewen opened the pull request:
[Cubiscan](https://cubiscan.com/) are dimensioners for cubing and weighing in
warehouses.  This module implements the communication with the dimensioners as
well as a screen to measure and weight packaging of the products.

Commit:
Add stock_cubiscan

gurneyalex committed:
[FIX] issues with stock_cubiscan

* was using a link to product.template in the wizard instead of product.product
* use product.packaging.type instead of the name of the packaging
* update product_packaging_type from OCA
* fix the numbe of digits of the volume

gurneyalex committed:
temporarily disable failing test

grindtildeath committed:
Integrate stock_cubiscan with product packaging types

damdam-s committed:
Add SSL support in cubiscan device

Commit:
Add setuptools files

Commit:
[IMP] stock_cubiscan: black, isort

Commit:
Migrate stock_cubiscan to 13.0

Commit:
Allow customizing CubiScan instantiation

Commit:
Add script to stub the Cubiscan server

Commit:
Add readme's source

Commit:
Add requirements

Commit:
run pre-commit with new prettiers

simahawk commented on stock_cubiscan/models/stock.py:
```diff
@@ -0,0 +1,25 @@
+
+class ProductPackaging(models.Model):
+    _inherit = "product.packaging"
+    # FIXME: move this in product_packaging_type
```
@guewen still the case?

guewen replied:
yes, it should be in this module IMO https://github.com/OCA/product-attribute/pull/536

simahawk replied:
can you update the comment w/ a link to https://github.com/OCA/product-attribute/tree/13.0/product_packaging_type ?

simahawk commented on stock_cubiscan/wizard/cubiscan_wizard.py:
```diff
@@ -0,0 +1,170 @@
+from odoo import _, api, fields, models
+
+
+class CubiscanWizard(models.TransientModel):
```
a docstring explaining what it does?

simahawk commented:
@guewen rebase/pre-commit needed.
@p-tombez pls pass by for a review ;)

p-tombez reviewed (commented):
LGTM, just need to fix the dependencies to make Travis happy

jgrandguillaume reviewed (approved):
Functional review ok on my side

guewen commented:
@p-tombez the dependency (https://github.com/OCA/web/pull/1480) has been merged, I amended the last commit to trigger travis and runbot builds

simahawk commented:
tests are :red_circle:

Commit:
Fix flaky test in cubiscan measures

The module product_packaging_type creates a packaging type "Default
Type" with sequence 1. In the test, the "unit" is created with sequence
1 too, which makes the ordering of the packagings in the test
"test_cubiscan_measures" undeterministic.

Note: "unit" should not be a packaging type, although the wizard should
allow to change the width, length and weight of the product too.

Commit:
Fix odoo warning because of duplicate field names

guewen commented:
@simahawk fixed

simahawk commented on stock_cubiscan/models/cubiscan.py:
```diff
@@ -0,0 +1,103 @@
+    def copy(self, default=None):
+        if not default:
+            default = {}
+        default["state"] = "not_ready"
```
wondering: couldn't we achieve the same by setting `copy=False` on the field?

simahawk reviewed (approved):
@guewen 2 little questions. LG anyway.

guewen commented:
@simahawk yes, you are right, I amended the migration commit with your 2 remarks

simahawk commented:
/ocabot merge
